Enyim.Caching/MurmurHash.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Enyim.Caching
+{
+    //-----------------------------------------------------------------------------
+    // MurmurHash3 was written by Austin Appleby, and is placed in the public
+    // domain. The author hereby disclaims copyright to this source code.
+    internal static class MurmurHash3
+    {
+        public static ulong Hash(string data)
+        {
+            if (string.IsNullOrEmpty(data))
+            {
+                throw new ArgumentNullException(nameof(data), "Input string cannot be null or empty.");
```

Comment:
We had modified this library to accept null/empty values. Will this negate that change?

Reply:
this is for keys not for values. We cant have a null key.